### ISSUE TITLE
Skip data prep when it's not needed

### ISSUE THREAD

**Issue by sashahafner** [AU-BCE-EE/ALFAM2#33]
With `prep = TRUE` but no relevant variables, `ALFAM2mod()` (can?) throw an error.

```
Error in `[.data.frame`(dat, , 1:ndum + ncc, drop = FALSE) :
  undefined columns selected

```

**Comment by sashahafner** [comment]
Fixed in v2.1.1 commit eb15e935cc018a284696c26775e5376569b1d0e8. Need to test, get into main, and make a release.

**Comment by sashahafner** [comment]
Seems to work well (with warning) around 68f50c033af348fb8e0e7c351533bdf4bbfabb4c
